src/function.ts

```diff
@@ -752,13 +758,23 @@ export class CallbackLambdaCode extends aws_lambda.Code {
                   // @ts-ignore - private - adds the tag manager, which we don't need
                   cfnProperties,
                   ...rest
-                } = transformTable(o as CfnResource);
+                } = transformBus(transformTable(o as CfnResource));
                 return transformTaggableResource(rest);
               } else if (Token.isUnresolved(o)) {
-                const token = (<any>o).toString();
-                // add to tokens to be turned into env variables.
-                tokens = [...tokens, token];
-                return token;
+                const reversed = Tokenization.reverse(o)!;
+                if (Reference.isReference(reversed)) {
+                  tokens.push(reversed.toString());
+                  return reversed.toString();
+                } else if (SecretValue.isSecretValue(reversed)) {
+                  throw new SynthError(
+                    ErrorCodes.Unsafe_use_of_secrets,
+                    "Found unsafe use of SecretValue token in a Function."
+                  );
```

Comment:
Any plans to plumb this through or is it impossible because it needs to be fetched from the secret store and may need to be regularly rotated, so better to do explicit?

Reply:
Have a ticket to plumb it through (the error code links to it). It should be possible, just need to turn it into our Secret object or similar. No reason we cannot send a hint with a the env variable that contains the ID

SSM: [id]
Secret: [id]

and then pull from secret manager during runtime. (also permissions)